zerver/views/user_settings.py

```diff
@@ -168,14 +168,18 @@ def json_set_avatar(request, user_profile):
         return json_error("You must upload exactly one avatar.")
 
     user_file = list(request.FILES.values())[0]
-    upload_avatar_image(user_file, user_profile, user_profile.email)
-    do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
-    user_avatar_url = avatar_url(user_profile)
-
-    json_result = dict(
-        avatar_url = user_avatar_url
-    )
-    return json_success(json_result)
+    try:
+        upload_avatar_image(user_file, user_profile, user_profile.email)
+        do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+
+        json_result = dict(
+            avatar_url = user_avatar_url
+        )
+        return json_success(json_result)
+    except IOError as e:
```

Comment:
IOError is the exception raised by PIL (see the [exception traceback](https://github.com/zulip/zulip/issues/583#issue-143342493)) (it would have been better if PIL had raised a better exception, we'll just have to deal with it). That's the most specific exception for me to catch. Maybe I can check the exception message and make it more specific.

The actual decoding happens in `upload_avatar_image_local`. I'll catch the exception there.

I don't fully understand what do you mean by raising JsonableError. Do you mean to say that I should check for IOError much deeper and raise JsonableError on catching it and then in the view, I'll catch JsonableError and return json_error?